File: binding/src/libqcheckbox.cpp
#include <QAbstractButton>
#include <QCheckBox>
#include <QEvent>
#include <QMouseEvent>
#include <QPaintEvent>
#include <QPoint>
#include <QSize>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <QStyleOptionButton>
#include <QWidget>
#include <qcheckbox.h>
#include "libqcheckbox.h"
#include "libqcheckbox.hxx"

QCheckBox* QCheckBox_new(QWidget* parent) {
	 return new VirtualQCheckBox(parent);
}

QCheckBox* QCheckBox_new2() {
	 return new VirtualQCheckBox();
}

QCheckBox* QCheckBox_new3(const libqt_string text) {
	 return new VirtualQCheckBox(QString::fromUtf8(text.data, text.len));
}

QCheckBox* QCheckBox_new4(const libqt_string text, QWidget* parent) {
	 return new VirtualQCheckBox(QString::fromUtf8(text.data, text.len), parent);
}

libqt_string QCheckBox_Tr(const char* s) {
	QString _ret = QCheckBox::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

QSize* QCheckBox_SizeHint(const QCheckBox* self) {
	return new QSize(self->sizeHint());
}

QSize* QCheckBox_MinimumSizeHint(const QCheckBox* self) {
	return new QSize(self->minimumSizeHint());
}

void QCheckBox_SetTristate(QCheckBox* self) {
	self->setTristate();
}

bool QCheckBox_IsTristate(const QCheckBox* self) {
	return self->isTristate();
}

int QCheckBox_CheckState(const QCheckBox* self) {
	return self->checkState();
}

void QCheckBox_SetCheckState(QCheckBox* self, int state) {
	self->setCheckState(static_cast<Qt::CheckState>(state));
}

void QCheckBox_StateChanged(QCheckBox* self, int param1) {
	self->stateChanged(param1);
}

void QCheckBox_Connect_StateChanged(QCheckBox* self, intptr_t slot) {
    void (*slotFunc)(QCheckBox*, int) = reinterpret_cast<void (*)(QCheckBox*, int)>(slot);
    QCheckBox::connect(self, &QCheckBox::stateChanged, [self, slotFunc](int param1) {
	slotFunc(self, param1);
    });
}

void QCheckBox_CheckStateChanged(QCheckBox* self, int param1) {
	self->checkStateChanged(static_cast<Qt::CheckState>(param1));
}

void QCheckBox_Connect_CheckStateChanged(QCheckBox* self, intptr_t slot) {
    void (*slotFunc)(QCheckBox*, int) = reinterpret_cast<void (*)(QCheckBox*, int)>(slot);
    QCheckBox::connect(self, &QCheckBox::checkStateChanged, [self, slotFunc](Qt::CheckState param1) {
	slotFunc(self, param1);
    });
}

libqt_string QCheckBox_Tr2(const char* s, const char* c) {
	QString _ret = QCheckBox::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QCheckBox_Tr3(const char* s, const char* c, int n) {
	QString _ret = QCheckBox::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QCheckBox_SetTristate1(QCheckBox* self, bool y) {
	self->setTristate(y);
}

// Base class handler implementation
QSize* QCheckBox_QBaseSizeHint(const QCheckBox* self) {
	auto* vqcheckbox = dynamic_cast<const VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
vqcheckbox->setQCheckBox_SizeHint_IsBase(true);
	return new QSize(vqcheckbox->sizeHint());
}
}

// Auxiliary method to allow providing re-implementation
void QCheckBox_OnSizeHint(const QCheckBox* self, intptr_t slot) {
	auto* vqcheckbox = dynamic_cast<const VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
vqcheckbox->setQCheckBox_SizeHint_Callback(reinterpret_cast<VirtualQCheckBox::QCheckBox_SizeHint_Callback>(slot));
}
}

// Base class handler implementation
QSize* QCheckBox_QBaseMinimumSizeHint(const QCheckBox* self) {
	auto* vqcheckbox = dynamic_cast<const VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
vqcheckbox->setQCheckBox_MinimumSizeHint_IsBase(true);
	return new QSize(vqcheckbox->minimumSizeHint());
}
}

// Auxiliary method to allow providing re-implementation
void QCheckBox_OnMinimumSizeHint(const QCheckBox* self, intptr_t slot) {
	auto* vqcheckbox = dynamic_cast<const VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
vqcheckbox->setQCheckBox_MinimumSizeHint_Callback(reinterpret_cast<VirtualQCheckBox::QCheckBox_MinimumSizeHint_Callback>(slot));
}
}

// Derived class handler implementation
bool QCheckBox_Event(QCheckBox* self, QEvent* e) {
	auto* vqcheckbox = dynamic_cast<VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
	return vqcheckbox->event(e);
	} else {
	return self->QCheckBox::event(e);
}
}

// Base class handler implementation
bool QCheckBox_QBaseEvent(QCheckBox* self, QEvent* e) {
	auto* vqcheckbox = dynamic_cast<VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
vqcheckbox->setQCheckBox_Event_IsBase(true);
	return vqcheckbox->event(e);
}
}

// Auxiliary method to allow providing re-implementation
void QCheckBox_OnEvent(QCheckBox* self, intptr_t slot) {
	auto* vqcheckbox = dynamic_cast<VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
vqcheckbox->setQCheckBox_Event_Callback(reinterpret_cast<VirtualQCheckBox::QCheckBox_Event_Callback>(slot));
}
}

// Derived class handler implementation
bool QCheckBox_HitButton(const QCheckBox* self, const QPoint* pos) {
	auto* vqcheckbox = dynamic_cast<const VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
	return vqcheckbox->hitButton(*pos);
	} else {
	return self->QCheckBox::hitButton(*pos);
}
}

// Base class handler implementation
bool QCheckBox_QBaseHitButton(const QCheckBox* self, const QPoint* pos) {
	auto* vqcheckbox = dynamic_cast<const VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
vqcheckbox->setQCheckBox_HitButton_IsBase(true);
	return vqcheckbox->hitButton(*pos);
}
}

// Auxiliary method to allow providing re-implementation
void QCheckBox_OnHitButton(const QCheckBox* self, intptr_t slot) {
	auto* vqcheckbox = dynamic_cast<const VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
vqcheckbox->setQCheckBox_HitButton_Callback(reinterpret_cast<VirtualQCheckBox::QCheckBox_HitButton_Callback>(slot));
}
}

// Derived class handler implementation
void QCheckBox_CheckStateSet(QCheckBox* self) {
	auto* vqcheckbox = dynamic_cast<VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
	vqcheckbox->checkStateSet();
	} else {
	self->QCheckBox::checkStateSet();
}
}

// Base class handler implementation
void QCheckBox_QBaseCheckStateSet(QCheckBox* self) {
	auto* vqcheckbox = dynamic_cast<VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
vqcheckbox->setQCheckBox_CheckStateSet_IsBase(true);
	vqcheckbox->checkStateSet();
}
}

// Auxiliary method to allow providing re-implementation
void QCheckBox_OnCheckStateSet(QCheckBox* self, intptr_t slot) {
	auto* vqcheckbox = dynamic_cast<VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
vqcheckbox->setQCheckBox_CheckStateSet_Callback(reinterpret_cast<VirtualQCheckBox::QCheckBox_CheckStateSet_Callback>(slot));
}
}

// Derived class handler implementation
void QCheckBox_NextCheckState(QCheckBox* self) {
	auto* vqcheckbox = dynamic_cast<VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
	vqcheckbox->nextCheckState();
	} else {
	self->QCheckBox::nextCheckState();
}
}

// Base class handler implementation
void QCheckBox_QBaseNextCheckState(QCheckBox* self) {
	auto* vqcheckbox = dynamic_cast<VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
vqcheckbox->setQCheckBox_NextCheckState_IsBase(true);
	vqcheckbox->nextCheckState();
}
}

// Auxiliary method to allow providing re-implementation
void QCheckBox_OnNextCheckState(QCheckBox* self, intptr_t slot) {
	auto* vqcheckbox = dynamic_cast<VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
vqcheckbox->setQCheckBox_NextCheckState_Callback(reinterpret_cast<VirtualQCheckBox::QCheckBox_NextCheckState_Callback>(slot));
}
}

// Derived class handler implementation
void QCheckBox_PaintEvent(QCheckBox* self, QPaintEvent* param1) {
	auto* vqcheckbox = dynamic_cast<VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
	vqcheckbox->paintEvent(param1);
	} else {
	self->QCheckBox::paintEvent(param1);
}
}

// Base class handler implementation
void QCheckBox_QBasePaintEvent(QCheckBox* self, QPaintEvent* param1) {
	auto* vqcheckbox = dynamic_cast<VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
vqcheckbox->setQCheckBox_PaintEvent_IsBase(true);
	vqcheckbox->paintEvent(param1);
}
}

// Auxiliary method to allow providing re-implementation
void QCheckBox_OnPaintEvent(QCheckBox* self, intptr_t slot) {
	auto* vqcheckbox = dynamic_cast<VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
vqcheckbox->setQCheckBox_PaintEvent_Callback(reinterpret_cast<VirtualQCheckBox::QCheckBox_PaintEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QCheckBox_MouseMoveEvent(QCheckBox* self, QMouseEvent* param1) {
	auto* vqcheckbox = dynamic_cast<VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
	vqcheckbox->mouseMoveEvent(param1);
	} else {
	self->QCheckBox::mouseMoveEvent(param1);
}
}

// Base class handler implementation
void QCheckBox_QBaseMouseMoveEvent(QCheckBox* self, QMouseEvent* param1) {
	auto* vqcheckbox = dynamic_cast<VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
vqcheckbox->setQCheckBox_MouseMoveEvent_IsBase(true);
	vqcheckbox->mouseMoveEvent(param1);
}
}

// Auxiliary method to allow providing re-implementation
void QCheckBox_OnMouseMoveEvent(QCheckBox* self, intptr_t slot) {
	auto* vqcheckbox = dynamic_cast<VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
vqcheckbox->setQCheckBox_MouseMoveEvent_Callback(reinterpret_cast<VirtualQCheckBox::QCheckBox_MouseMoveEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QCheckBox_InitStyleOption(const QCheckBox* self, QStyleOptionButton* option) {
	auto* vqcheckbox = dynamic_cast<const VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
	vqcheckbox->initStyleOption(option);
	} else {
	self->QCheckBox::initStyleOption(option);
}
}

// Base class handler implementation
void QCheckBox_QBaseInitStyleOption(const QCheckBox* self, QStyleOptionButton* option) {
	auto* vqcheckbox = dynamic_cast<const VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
vqcheckbox->setQCheckBox_InitStyleOption_IsBase(true);
	vqcheckbox->initStyleOption(option);
}
}

// Auxiliary method to allow providing re-implementation
void QCheckBox_OnInitStyleOption(const QCheckBox* self, intptr_t slot) {
	auto* vqcheckbox = dynamic_cast<const VirtualQCheckBox*>(self);
	if (vqcheckbox && vqcheckbox->isVirtualQCheckBox) {
vqcheckbox->setQCheckBox_InitStyleOption_Callback(reinterpret_cast<VirtualQCheckBox::QCheckBox_InitStyleOption_Callback>(slot));
}
}

void QCheckBox_Delete(QCheckBox* self) {
    delete self;
}

File: binding/include/libqcheckbox.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qcheckbox.html
pub const qcheckbox = struct {

    /// New constructs a new QCheckBox object.
    pub fn New(parent: ?*anyopaque) QtC.QCheckBox {
        return qtc.QCheckBox_new(@ptrCast(parent));
    }


    /// New2 constructs a new QCheckBox object.
    pub fn New2() QtC.QCheckBox {
        return qtc.QCheckBox_new2();
    }


    /// New3 constructs a new QCheckBox object.
    pub fn New3(text: []const u8) QtC.QCheckBox {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
return qtc.QCheckBox_new3(text_str);
    }


    /// New4 constructs a new QCheckBox object.
    pub fn New4(text: []const u8, parent: ?*anyopaque) QtC.QCheckBox {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
return qtc.QCheckBox_new4(text_str, @ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCheckBox_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcheckbox.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QCheckBox_SizeHint(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnSizeHint(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) QtC.QSize) void {
        qtc.QCheckBox_OnSizeHint(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QCheckBox_QBaseSizeHint(@ptrCast(self));
    }

    pub fn MinimumSizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QCheckBox_MinimumSizeHint(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnMinimumSizeHint(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) QtC.QSize) void {
        qtc.QCheckBox_OnMinimumSizeHint(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseMinimumSizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QCheckBox_QBaseMinimumSizeHint(@ptrCast(self));
    }

    pub fn SetTristate(self: ?*anyopaque, ) void {
        qtc.QCheckBox_SetTristate(@ptrCast(self));
    }

    pub fn IsTristate(self: ?*anyopaque, ) bool {
        return qtc.QCheckBox_IsTristate(@ptrCast(self));
    }

    pub fn CheckState(self: ?*anyopaque, ) i32 {
        return qtc.QCheckBox_CheckState(@ptrCast(self));
    }

    pub fn SetCheckState(self: ?*anyopaque, state: i32) void {
        qtc.QCheckBox_SetCheckState(@ptrCast(self), @intCast(state));
    }

    pub fn StateChanged(self: ?*anyopaque, param1: i32) void {
        qtc.QCheckBox_StateChanged(@ptrCast(self), param1);
    }

    pub fn OnStateChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QCheckBox_Connect_StateChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn CheckStateChanged(self: ?*anyopaque, param1: i32) void {
        qtc.QCheckBox_CheckStateChanged(@ptrCast(self), @intCast(param1));
    }

    pub fn OnCheckStateChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QCheckBox_Connect_CheckStateChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Event(self: ?*anyopaque, e: ?*anyopaque) bool {
        return qtc.QCheckBox_Event(@ptrCast(self), @ptrCast(e));
    }

    /// Allows for overriding the related default method
    pub fn OnEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) bool) void {
        qtc.QCheckBox_OnEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseEvent(self: ?*anyopaque, e: ?*anyopaque) bool {
        return qtc.QCheckBox_QBaseEvent(@ptrCast(self), @ptrCast(e));
    }

    pub fn HitButton(self: ?*anyopaque, pos: ?*anyopaque) bool {
        return qtc.QCheckBox_HitButton(@ptrCast(self), @ptrCast(pos));
    }

    /// Allows for overriding the related default method
    pub fn OnHitButton(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) bool) void {
        qtc.QCheckBox_OnHitButton(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseHitButton(self: ?*anyopaque, pos: ?*anyopaque) bool {
        return qtc.QCheckBox_QBaseHitButton(@ptrCast(self), @ptrCast(pos));
    }

    pub fn CheckStateSet(self: ?*anyopaque, ) void {
        qtc.QCheckBox_CheckStateSet(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnCheckStateSet(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCheckBox_OnCheckStateSet(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseCheckStateSet(self: ?*anyopaque, ) void {
        qtc.QCheckBox_QBaseCheckStateSet(@ptrCast(self));
    }

    pub fn NextCheckState(self: ?*anyopaque, ) void {
        qtc.QCheckBox_NextCheckState(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnNextCheckState(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCheckBox_OnNextCheckState(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseNextCheckState(self: ?*anyopaque, ) void {
        qtc.QCheckBox_QBaseNextCheckState(@ptrCast(self));
    }

    pub fn PaintEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QCheckBox_PaintEvent(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnPaintEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QCheckBox_OnPaintEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBasePaintEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QCheckBox_QBasePaintEvent(@ptrCast(self), @ptrCast(param1));
    }

    pub fn MouseMoveEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QCheckBox_MouseMoveEvent(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnMouseMoveEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QCheckBox_OnMouseMoveEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseMouseMoveEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QCheckBox_QBaseMouseMoveEvent(@ptrCast(self), @ptrCast(param1));
    }

    pub fn InitStyleOption(self: ?*anyopaque, option: ?*anyopaque) void {
        qtc.QCheckBox_InitStyleOption(@ptrCast(self), @ptrCast(option));
    }

    /// Allows for overriding the related default method
    pub fn OnInitStyleOption(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QCheckBox_OnInitStyleOption(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseInitStyleOption(self: ?*anyopaque, option: ?*anyopaque) void {
        qtc.QCheckBox_QBaseInitStyleOption(@ptrCast(self), @ptrCast(option));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCheckBox_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcheckbox.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCheckBox_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcheckbox.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetTristate1(self: ?*anyopaque, y: bool) void {
        qtc.QCheckBox_SetTristate1(@ptrCast(self), y);
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QCheckBox_Delete(@ptrCast(self));
    }
};
